main.cpp

```diff
@@ -979,6 +981,162 @@ void SelfTest(bool verbose) {
 
 //----------------------------------------------------------------------------
 
+// These probability tables for weighted average exist under assumption that hash speed does not depend
```

Comment:
Aren't those tables way over the top?
The rest looks good

Reply:
@rurban what would be the better option? I see following ones:
- keep tables as they're fairly small and somewhat easy to understand, but add a better comment and/or better representation. E.g. histogram with bins having integer values instead of probabilities.
- keep the distribution tabulated, but make it an external parameter, passed via environment and/or file
- replace the tables with some function, I suspect that these distributions are possible to approximate with a reasonable accuracy
- drop the weighted average from SpeedSmall, move that to some kind of wrapper script and forget about the idea of a "real-world" average at #113

Reply:
I personally prefer tabulated distribution passed as an environment variable. It keeps things simple and does not place anyone into the shoes of a curator for a dataset of "representative" hashed strings.